extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -0,0 +1,241 @@
+  );
+
+  useLayoutEffect(() => {
+    const sidebarElement = sidebarRef.current;
+    const maskElement = maskRef.current;
+    if (!sidebarElement || !maskElement) {
+      return;
+    }
+
+    let sidebarAnimation;
+    let maskAnimation;
+    // "Make Visible" Animation
+    if (opened) {
+      const postVisibleAnim = () => {
+        safelySetStyles(sidebarElement, ANIMATION_STYLES_SIDEBAR_FINAL);
+        safelySetStyles(maskElement, ANIMATION_STYLES_MASK_FINAL);
+      };
+      if (!sidebarElement.animate || !maskElement.animate) {
+        postVisibleAnim();
+        return;
+      }
+
+      safelySetStyles(
+        sidebarElement,
+        side === 'left'
+          ? ANIMATION_STYLES_SIDEBAR_LEFT_INIT
+          : ANIMATION_STYLES_SIDEBAR_RIGHT_INIT
+      );
+      safelySetStyles(maskElement, ANIMATION_STYLES_MASK_INIT);
+      sidebarAnimation = sidebarElement.animate(
+        side === 'left'
+          ? ANIMATION_KEYFRAMES_SLIDE_IN_LEFT
+          : ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT,
+        {
+          duration: ANIMATION_DURATION,
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postVisibleAnim;
+      maskAnimation = maskElement.animate(ANIMATION_KEYFRAMES_FADE_IN, {
+        duration: ANIMATION_DURATION,
+        fill: 'both',
+        easing: ANIMATION_EASE_IN,
+      });
+    } else {
+      // "Make Invisible" Animation
+      const postInvisibleAnim = () => {
+        if (onAfterCloseRef.current) {
+          onAfterCloseRef.current();
+        }
+        sidebarAnimation = null;
+        maskAnimation = null;
+        setMounted(false);
+      };
+      if (!sidebarElement.animate || !maskElement.animate) {
+        postInvisibleAnim();
+        return;
+      }
+      sidebarAnimation = sidebarElement.animate(
+        side === 'left'
+          ? ANIMATION_KEYFRAMES_SLIDE_IN_LEFT
+          : ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT,
+        {
+          duration: ANIMATION_DURATION,
+          direction: 'reverse',
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postInvisibleAnim;
+      maskAnimation = maskElement.animate(ANIMATION_KEYFRAMES_FADE_IN, {
+        duration: ANIMATION_DURATION,
+        direction: 'reverse',
+        fill: 'both',
+        easing: ANIMATION_EASE_IN,
+      });
+    }
+    return () => {
+      if (sidebarAnimation) {
+        sidebarAnimation.cancel();
+      }
+      if (maskAnimation) {
+        maskAnimation.cancel();
+      }
+    };
+  }, [opened, onAfterCloseRef, side]);
+
+  return (
+    mounted && (
+      <>
+        <ContainWrapper
+          as={Comp}
+          ref={sidebarRef}
+          size={false}
+          layout={true}
+          paint={true}
+          className={`${classes.sidebarClass} ${
+            side === 'left' ? classes.left : classes.right
+          }`}
+          role="menu"
+          tabindex="-1"
```

Comment:
Isn't a `<div>` not focusable by default?

Reply:
Also, shouldn't it be focusable so it can listen for `Esc` to close?

Reply:
The only ways to close that have been implemented so far are the imperative API and click on the mask.  I have an action item in the tracker: https://github.com/ampproject/amphtml/issues/31366 for handling focus, so I'll follow up at that point :).  This may be updated.